### PR TITLE
Refactor and reorder domain suggestions in domain picker

### PR DESCRIPTION
**Pull request by alshakero**
This is a new fresh PR that is similar to https://github.com/Automattic/wp-calypso/pull/42011.

It reorders domains suggestions in domain picker as follows:
> recommended
free
the rest

Fixes: https://github.com/Automattic/wp-calypso/issues/42010

### CODE REVIEW

File: client/landing/gutenboarding/utils/domain-suggestions.ts [hunk 1]
```diff
@@ -1,37 +1,5 @@
-/**
- * External dependencies
- */
-import type { DomainSuggestions } from '@automattic/data-stores';
-
-type DomainSuggestion = DomainSuggestions.DomainSuggestion;
+type DomainSuggestion = import('@automattic/data-stores').DomainSuggestions.DomainSuggestion;
```
Comment (razvanpapadopol) [inline client/landing/gutenboarding/utils/domain-suggestions.ts:1]: We moved away from importing types like this in [#42804](https://github.com/Automattic/wp-calypso/pull/42804).
See this exact type import change:
https://github.com/Automattic/wp-calypso/pull/42804/files#diff-9fb3f50c91d02f132bfab858e87cb792R10-R35

File: packages/domain-picker/src/utils/domain-suggestions.ts [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-			return suggestion;
-		}
-		if ( suggestion.relevance > result.relevance ) {
-			return suggestion;
-		}
-		return result;
-	} );
-
-	return recommendedSuggestion;
-}
```
Comment (razvanpapadopol) [inline packages/domain-picker/src/utils/domain-suggestions.ts:31]: ❤️

File: packages/domain-picker/src/domain-picker/index.tsx [hunk 3]
```diff
@@ -89,69 +83,43 @@ const DomainPicker: FunctionComponent< Props > = ( {
...
-	// 		setCurrentSelection( currentDomain );
-	// 		return;
-	// 	}
 
-	// 	// Recalculate free-domain suggestions inside the closure. `getFreeDomainSuggestions()`
-	// 	// always returns a new object so it shouldn't be used in `useEffects()` dependencies list.
-	// 	const latestFreeSuggestion = getFreeDomainSuggestions( allSuggestions );
+	// the API puts the subdomain on top, push it to second place
+	if ( domainSuggestions && domainSuggestions.length > 1 ) {
+		domainSuggestions?.splice( 0, 2, domainSuggestions[ 1 ], domainSuggestions[ 0 ] );
```
Comment (daledupreez) [inline packages/domain-picker/src/domain-picker/index.tsx]: This feels fragile to me. I'd still prefer to have this extract the free options using `is_free` and then re-insert those results after the first position.

Comment (alshakero) [reply packages/domain-picker/src/domain-picker/index.tsx]: Yes I changed it to that. Thanks!

**Pull request merged**